### PR TITLE
Fix modelsave

### PR DESCRIPTION
**Pull request by YujiOshima**
Fix bug that the last trial of the study won't be saved when a study has completed.
I changed to save models of all completed trials before breaking `trialIteration`.

### CODE REVIEW

**Comment by gaocegege** [comment]
/assign @mitake  

For the changes in dlk.

File: dlk/dlkmanager/learningTask.go [hunk 1]
```diff
@@ -307,12 +307,12 @@ func (lt *learningTask) run() {
 			}
 
 		case <-time.After(1 * time.Second):
-			lt.pollJobs()
 			err := lt.checkPodStatus(podState)
 			if err != nil {
 				fmt.Println(err.Error())
 				os.Exit(1)
 			}
+			lt.pollJobs()
```
[the remark this answers is not in the document]
Comment (YujiOshima) [reply dlk/dlkmanager/learningTask.go]: If pollJobs is called before checkPodStatus, when logs of pod updated and the pod completed after pollJobs was called, the last logs won't be collected by dlk.

[the remark this answers is not in the document]
Comment (mitake) [reply dlk/dlkmanager/learningTask.go]: I understood, thanks!

**Comment by mitake** [comment]
@gaocegege the change of dlk side looks good to me

**Comment by gaocegege** [comment]
/lgtm

@mitake Thanks :-)

**Pull request merged**